pkgs/builtfilter-rs/default.nix

```diff
@@ -0,0 +1,27 @@
+{
+  inputs,
+  pkgs,
+  lib,
+  self,
+  ...
+}: let
+  cargoToml = lib.importTOML "${self}/builtfilter-rs/Cargo.toml";
+in
+  pkgs.rustPlatform.buildRustPackage
+  rec
+  {
+    pname = cargoToml.package.name;
+    version = cargoToml.package.version;
+    cargoLock = {
+      lockFile = src + "/Cargo.lock";
+      allowBuiltinFetchGit = true;
+    };
+    src = self + "/builtfilter-rs";
```

Comment:
does it not improve caching if we use `../../builtfilter-rs` as it wouldn't depend on the source of the repo?
with `self` i think we'll need to rebuild this binary whenever anything in the repo changes. There might be something smart in `buildRustPackage` that I forgot about, but if not that's likely costing us with every build.

@tomberek